crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -0,0 +1,134 @@
+use super::Primitive2d;
+use crate::Vec2;
+
+/// A normalized vector pointing in a direction in 2D space
+#[derive(Clone, Copy, Debug)]
+pub struct Direction2d(Vec2);
```

Comment:
Apart from `Quad` (quadrilateral), it'd be the only abbreviated primitive, but it does resemble a struct like `Vec2` more than an actual primitive shape... if it will be used *a lot* in public APIs, then perhaps `Dir2d` would be good, but otherwise I think I prefer `Direction2d` since it's generally better not to abbreviate names in my eyes. Other opinions on this would be useful though

Reply:
> it's generally better not to abbreviate

Sure but have you been looking at Rust, didn't that ship float already? 😅

Reply:
For very common types that appear everywhere like `Vec`, `Vec2`, `Quat` and `Res`, it makes sense to abbreviate, but for most other types I don't consider it good practise. But `Dir2d` could be considered to be a utility type similar to `Vec2`, so it could work here, not sure. (although we have the extra `d` there... `Dir2` would be more consistent with Glam math types, but not with Bevy's other types)

Reply:
Yes it's difficult to tell in advance. I don't mind either way.

Reply:
I'd prefer to be consistent and explicit here. I don't think it's common enough to warrant shortening.

Reply:
I’d vote Direction2d for what it’s worth.

Reply:
If we use `Normalized<T>(T)`, it could be used for T = `Vec2`, `Vec3`, `Quat`, etc. Aliases could be defined as well.

I think it's okay to abbreviate often-used types, such as `Vec2`/`Vec3` (integers and floats only get a single letter), but using an expanded name for less common types avoids confusion and time spent searching documentation.

Reply:
IMO `Direction2d` is better because it is more explicit. Since most users are usually using `rust-analyzer`, you can just type `dir2` and the fuzzy matching autocomplete will do the work for you, so you would type less anyways.

![image](https://github.com/bevyengine/bevy/assets/1176452/bdbf2da4-125b-4f79-9ef2-0190e60ff85b)

Reply:
Yeah we can go for `Direction2d` since it's very explicit and consistent with other names. I kinda like the `Normalized` suggestion too, but we only need it for `Vec2`/`Vec3` for now, so I think it's fine if we go for the non-generic approach until there's a real need for it. It would probably be aliased under `Direction2d`/`Direction3d` anyway and the API would be basically the same, so changing it later if necessary should be pretty straightforward.